robot/impl/resource_manager.go

```diff
@@ -211,7 +213,22 @@ func (manager *resourceManager) updateRemoteResourceNames(
 		if _, alreadyCurrent := activeResourceNames[resName]; alreadyCurrent {
 			activeResourceNames[resName] = true
 			if ok && !gNode.IsUninitialized() {
-				continue
+				if !recreateAllClients {
```

Comment:
Could we leave a comment above this line saying what situation `ok && !gNode.IsUnitialized` represents and why we continue with `!recreateAllClients`, and mark children for update + close the node if `recreateAllClients`?